bin/.upgrades/versions/v1/upgradeStorage.js

```diff
@@ -0,0 +1,389 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const
+  _ = require('lodash'),
+  getESConnector = require('../../connectors/es');
+
+const
+  INTERNAL_PREFIX = '%',
+  PUBLIC_PREFIX = '&',
+  NAME_SEPARATOR = '.';
+
+function getNewIndexName (index, collection) {
+  const prefix = index[0] === '%' ? '' : PUBLIC_PREFIX;
+
+  return `${prefix}${index}${NAME_SEPARATOR}${collection}`;
+}
+
+async function moveData (source, target, index, collection, newIndex) {
+  let page = await source.search({
+    index,
+    type: collection,
+    body: { sort: [ '_doc' ] },
+    scroll: '1m',
+    size: 1000
+  });
+
+  const total = page.body.hits.total;
+  let moved = 0;
+
+  while (moved < total) {
+    const bulk = [];
+
+    for (let i = 0; i < page.body.hits.hits.length; i++) {
+      const doc = page.body.hits.hits[i];
+
+      if (doc._source._kuzzle_info) {
+        delete doc._source._kuzzle_info.active;
+        delete doc._source._kuzzle_info.deletedAt;
+      }
+
+      bulk.push({ index: { _index: newIndex, _id: doc._id } });
+      bulk.push(doc._source);
+    }
+
+    await target.bulk({ body: bulk, _source: false });
+
+    moved += page.body.hits.hits.length;
+
+    if (moved < total) {
+      page = await source.scroll({
+        scroll_id: page.body._scroll_id,
+        scroll: '1m'
+      });
+    }
+  }
+
+  return total;
+}
+
+async function upgradeMappings (source, target, index, collection, newIndex) {
+  const
+    mappingsResponse = await source.indices.getMapping({
+      index,
+      type: collection
+    }),
+    mappings = mappingsResponse.body[index].mappings[collection];
+
+  // remove obsolete mapping properties
+  if (mappings.properties && mappings.properties._kuzzle_info) {
+    delete mappings.properties._kuzzle_info.properties.active;
+    delete mappings.properties._kuzzle_info.properties.deletedAt;
+  }
+
+  await target.indices.putMapping({
+    index: newIndex,
+    body: {
+      properties: mappings.properties,
```

Comment:
Here we should handle the `_kuzzle_info` separately because we have record of clients using an incorrect version of Kuzzle Metadata mappings (`text` instead of `keyword`).  
A solution could be to try to detect if Kuzzle Metadata mappings are different than the default ones and if so propose to the user to change mappings for the default ones

Reply:
Hm... Good catch. And that's a tricky one: I'll give it some thought after I finish the cache migration, but my first instinct is to automatically (and silently) convert the mapping: only the `author` field should be impacted, and in older versions of Kuzzle it was set to `text` because, well, we let ES set the mappings for kuzzle's metadata automatically. 
This time is as good as any to fix that error definitely.